### PR TITLE
Korvus x Firecrawl blog post

### PR DESCRIPTION
**Pull request by SilasMarvin**

### CODE REVIEW

File: pgml-cms/blog/fire.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+2. Crawls the website using the `crawl()` function.
+3. Constructs a list of documents from the crawled data, using the source URL as the ID and the markdown content as the document text.
+4. Upserts these documents into the collection. The pipeline automatically splits the markdown and generates embeddings for each chunk storing it all in Postgres.
+
+### Performing RAG
+
+With our data indexed, we can now perform RAG:
+
+```python
+async def do_rag(user_query):
```
Comment (montanalow) [inline pgml-cms/blog/fire.md:144]: This function name cracks me up. https://en.wikipedia.org/wiki/Durag#Spelling_and_etymology

File: pgml-cms/blog/fire.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+```python
+from korvus import Collection, Pipeline
+from firecrawl import FirecrawlApp
+import os
+import time
+import asyncio
+from rich import print
+
+# Initialize the FirecrawlApp with your API key
+app = FirecrawlApp(api_key=os.environ["FIRECRAWL_API_KEY"])
```
Comment (montanalow) [inline pgml-cms/blog/fire.md:49]: I might name this variable `firecrawl` for clarity.

File: pgml-cms/blog/fire.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+async def main():
+    # ... (previous code for setup and indexing)
+
+    # Now we can search
+    while True:
+        user_query = input("\n\nquery > ")
+        if user_query == "q":
+            break
+        results = await do_rag(user_query)
+        print(results)
```
Comment (montanalow) [inline pgml-cms/blog/fire.md:213]: Good opportunity to show off some fun example results, and compare it to llama 3.1 w/o RAG.

File: pgml-cms/blog/fire.md (hunk 1 above)
Comment (SilasMarvin) [reply pgml-cms/blog/fire.md:144]: haha I know its hilarious

**Pull request merged**